libs/src/services/creatorNode/index.js

```diff
@@ -339,6 +345,57 @@ class CreatorNode {
     return body
   }
 
+  /**
+   * Track upload flow with polling.
+   * @param {File} file track to upload
+   * @param {function?} onProgress called with loaded bytes and total bytes
+   * @return {Object} response body
+   *
+   * @note Rename this function to `uploadTrackAudio` and update the actual route when
+   * we want to fully use polling for track content, and remove the existing method
+   */
+  async uploadTrackAudioPolling (file, onProgress) {
+    let uuid
+    try {
+      ({ data: { uuid } } = await this._uploadFile(file, '/track_content_async', onProgress))
+    } catch (e) {
+      await this._handleErrorHelper(e)
+    }
+
+    const start = Date.now()
+    while (Date.now() - start < MAX_TRACK_TRANSCODE_TIMEOUT) {
+      const { status, resp } = await this.getProcessingStatus('transcode', uuid)
+      // Should have a body structure of:
+      //   { transcodedTrackCID, transcodedTrackUUID, track_segments, source_file }
+      if (status && status === 'DONE') return resp
+      if (status && status === 'FAILED') await this._handleErrorHelper(new Error(`Transcode failed: uuid=${uuid}, error=${resp}`), `/track_content`, uuid)
+
+      // Check the transcode status every 5s
+      await wait(5000)
```

Comment:
i wonder if 5s is the right number

Reply:
yeah, this number was arbitrary. can adjust if anyone feels strongly

Reply:
my gut says 2-3s, i dont think the load implication will be all that bad

Reply:
updating to 3s

Reply:
we can just make this 1s to start and see how that goes?